js/pivx_shielding.js

```diff
@@ -258,6 +274,7 @@ export class PIVXShielding {
    */
   discardTransaction(txid) {
     this.pendingSpentNotes.clear(txid);
+    this.pendingUnspentNotes.clear(txid);
```

Comment:
This means the unspent notes will be cleared when calling finalizeTransaction too